[Bug Fix] APAtrap cast output to string before parsing and avoid time out error

faricazjj opened the pull request:
Fixes #402 

This PR solves two errors encountered when running APAtrap with real data.

1. When converting APAtrap output to absolute TPM quantification challenge output bed file, a column containing a list of TPMs is taken and parsed by splitting by comma. This column is known to contain strings and so is directly split by comma, which has worked for most of the datasets. 

However, while running APAtrap on one of the Keratinocyte samples, one row was written as int in the APAtrap output file while the rest were written as string as expected. Hence, this row caused the workflow to error out with the following message:
![Screen Shot 2022-08-03 at 9 04 40 PM](https://user-images.githubusercontent.com/25573986/183491039-a490b592-f4e6-4953-b97b-3c3c6836cc32.png)

A solution for this is to always cast the rows to string before splitting as such:

![Screen Shot 2022-08-08 at 2 49 42 PM](https://user-images.githubusercontent.com/25573986/183491887-3bc92282-655f-4443-b53d-dec6323c237a.png)


2. APAtrap takes a long time when processing real datasets. Even when processing the samples per chromosome, it could take at least 12 hours depending on the number of samples in the sample sheet. As such, APAtrap is prone to time out errors or out of memory errors. One solution is to add the label 'process_high' in APAtrap processes to give it more time to finish running to avoid time out errors.



## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have updated corresponding READMEs (if applicable)
- [x] My code follows the templates/style guidelines of the repository
- [x] In- and output formats comply with APAeval specifications
- [x] No parameters or file names are hardcoded
- [x] Results, logs or other output is not commited to the repository
- [x] I have tested my code

yuukiiwa reviewed (approved):
Thanks, Farica! Looks good to me!!

dominikburri reviewed (commented):
Hi @faricazjj, many thanks for the fixes! 
I successfully tested it with the provided test data and am quite sure your fixes work for the real data you mention.

I'm kind of nitpicking now.  The _process_high_ label uses by default 12 cpus, but the individual processes use only 1 cpu as I gauge from the code. Maybe you can use a label like this:
```
withLabel:process_big_mem {
    cpus   = { check_max( 1 * task.attempt, 'cpus' ) }
    memory = { check_max( 128.GB * task.attempt, 'memory' ) }
    time   = { check_max( 72.h * task.attempt, 'time' ) }
  }
```
Also, not sure whether all 3 processes need the _process_high_ label. Maybe only one process is doing the heavy computation?
I'm just trying to tailor the execution, such that resources are sensibly used.

grexor reviewed (dismissed):
looks good

faricazjj commented:
> Hi @faricazjj, many thanks for the fixes! I successfully tested it with the provided test data and am quite sure your fixes work for the real data you mention.
> 
> I'm kind of nitpicking now. The _process_high_ label uses by default 12 cpus, but the individual processes use only 1 cpu as I gauge from the code. Maybe you can use a label like this:
> 
> ```
> withLabel:process_big_mem {
>     cpus   = { check_max( 1 * task.attempt, 'cpus' ) }
>     memory = { check_max( 128.GB * task.attempt, 'memory' ) }
>     time   = { check_max( 72.h * task.attempt, 'time' ) }
>   }
> ```
> 
> Also, not sure whether all 3 processes need the _process_high_ label. Maybe only one process is doing the heavy computation? I'm just trying to tailor the execution, such that resources are sensibly used.

@dominikburri thank you for the suggestion! I initially thought these are the cap for the resources but after seeing your comment and also [nextflow documentation](https://nf-co.re/usage/configuration) these are actually allocated resources. Thanks for being meticulous! :D I've updated the PR as suggested and added in the README a note on it!

dominikburri reviewed (approved):
Thank you @faricazjj! LGTM :)

yuukiiwa reviewed (commented):
Thanks, @faricazjj!! Looks good to me!!
